libpng/CMakeLists.txt

```diff
@@ -0,0 +1,6 @@
+set(COMPONENT_ADD_INCLUDEDIRS
+              ./ libpng )
+set(COMPONENT_SRCDIRS ./libpng )
+set(COMPONENT_REQUIRES )
+set(COMPONENT_PRIV_REQUIRES zlib)
+register_component()
```

Comment:
```suggestion
idf_component_register(SRCDIRS ./libpng INCLUDE_DIRS ./ libpng)
```

Reply:
thanks, but libpng library depends on zlib library

Reply:
In this case you are already specifying the dependency on zlib using idf_component.yml file. The build system will automatically add the name of the component (`espressif__zlib`) to the dependency list, so `zlib` doesn't need to be listed here.

Reply:
@shootao Could you please convert this code to use `idf_component_register` syntax, similar to what is suggested above? `register_component` syntax is from IDF v3.3, which does not support the IDF Component Manager and is already EoL.

Also, seems like you still have the `zlib` listed as a dependency here. Please see my comment above, this dependency will be added automatically by the component manager, we don't need to specify it in CMake.